### PR TITLE
[WIP] Julia 1.0.0

### PR DESCRIPTION
**Pull request by scls19fr**
It's building
It's also working for my use case 
but don't merge before discussing about these changes

### CODE REVIEW

File: src/wrap.jl [hunk 1]
```diff
@@ -428,40 +428,40 @@ end
...
+                (Config, Ref{Int32}), config, cts)
     handle_error(ret, loc())
     cts[]
 end
 
 # enum sp_return sp_set_config_cts(struct sp_port_config *config, enum sp_cts cts);
 function sp_set_config_cts(config::Config, cts::SPcts)
     ret = ccall((:sp_set_config_cts, libserialport), SPReturn,
-                (Config, SPcts), config, SPcts(cts))
+                (Config, Int32), config, Int(cts))
```
Comment (scls19fr) [inline src/wrap.jl]: This is the kind of changes I have to do... but that I don't like

File: src/wrap.jl [hunk 2]
```diff
@@ -428,40 +428,40 @@ end
...
     handle_error(ret, loc())
     ret
 end
 
 # enum sp_return sp_set_dtr(struct sp_port *port, enum sp_dtr dtr);
 function sp_set_dtr(port::Port, dtr::SPdtr)
     ret = ccall((:sp_set_dtr, libserialport), SPReturn,
-                (Port, SPdtr), port, dtr)
+                (Port, Int32), port, Int(dtr))
```
Comment (scls19fr) [inline src/wrap.jl]: same here...

Should it be Int32, Int8, Int16, Int64...

I'm pretty sure that we should be more "generic", getting underlying type of SPdtr at compile time.


I also did:

    Int(dtr)

but I should have done

    Integer(dtr)

For your information

```
julia> typeof(Int(SP_DTR_ON))
Int64

julia> typeof(Integer(SP_DTR_ON))
Int32
```

**Comment by andrewadare** [comment]
This is to address #17. 

There are some changes that would be useful to include immediately, such as removing tic/toc, but I am less comfortable with other changes, such as casting the libserialport types to integers of various widths.

I'm going to leave this open for now, bringing in some of your changes manually (so this PR still definitely has value). We'd certainly like to be sure everything works properly with Julia 1.0 on 32 bit hosts such as a Raspberry Pi as well as on 64 bit systems.

**Comment by scls19fr** [comment]
What we need to avoid to smear code like I did, is to be able to get the underlying type of an Enum.
Maybe @tanmaykm could give us a tip on this?

**Comment by andrewadare** [comment]
> What we need to avoid to smear code like I did, is to be able to get the underlying type of an Enum.

I think adding constructors for the Enums in 03a0ff516ca8e430b8c2e3faf1ecc2cb5249fd3f was sufficient to deal with the deprecation warnings/errors.

**Comment by scls19fr** [comment]
Great idea! However I'm still wondering if the underlying type of an Enum can be known (for some others projects)

**Comment by andrewadare** [comment]
I may not be understanding what you're trying to do, but according to https://docs.julialang.org/en/stable/base/base/#Special-Types-1, the base type is Int32 by default:

> BaseType, which defaults to Int32, must be a primitive subtype of Integer. Member values can be converted between the enum type and BaseType. read and write perform these conversions automatically.

**Comment by andrewadare** [comment]
This is kind of a silly example, but demonstrates the point:
```julia
using LibSerialPort

ret = SPReturn

for instance in instances(ret)
       println("$(typeof(instance)) $(supertype(typeof(instance)))")
end
```

```
SPReturn Enum{Int32}
SPReturn Enum{Int32}
SPReturn Enum{Int32}
SPReturn Enum{Int32}
SPReturn Enum{Int32}
```

**Comment by scls19fr** [comment]
@enum Fruit::UInt8 apple orange kiwi

is allowed but how to get UInt8 from Fruit?

**Comment by andrewadare** [comment]
This is apparently one way:
```
julia> supertype(Fruit).parameters[1]
UInt8
```

**Comment by tanmaykm** [comment]
Another way could be:
```
julia> Base.Enums.basetype(Fruit)
UInt8
```

**Comment by scls19fr** [comment]
Thanks @tanmaykm however I opened an issue at https://github.com/JuliaLang/julia/issues/28778 because I think there is a room for improvement (at least for doc)

**Pull request closed without merging**